lib/web_ui/lib/src/engine/text/canvas_paragraph.dart

```diff
@@ -157,85 +157,44 @@ class CanvasParagraph implements ui.Paragraph {
     cssStyle
       ..position = 'absolute'
       // Prevent the browser from doing any line breaks in the paragraph. We want
-      // to insert our own <BR> breaks based on layout results.
+      // to have full control of the paragraph layout.
       ..whiteSpace = 'pre';
 
-    if (width > longestLine) {
-      // In this case, we set the width so that the CSS text-align property
-      // works correctly.
-      // When `longestLine` is >= `paragraph.width` that means the DOM element
-      // will automatically size itself to fit the longest line, so there's no
-      // need to set an explicit width.
-      cssStyle.width = '${width}px';
-    }
-
-    if (paragraphStyle.maxLines != null || paragraphStyle.ellipsis != null) {
-      cssStyle
-        ..overflowY = 'hidden'
-        ..height = '${height}px';
-    }
-
     // 2. Append all spans to the paragraph.
 
-    FlatTextSpan? span;
-
-    html.HtmlElement element = rootElement;
+    html.HtmlElement? lastSpanElement;
     final List<EngineLineMetrics> lines = computeLineMetrics();
 
     for (int i = 0; i < lines.length; i++) {
-      // Insert a <BR> element before each line except the first line.
-      if (i > 0) {
-        element.append(html.document.createElement('br'));
-      }
-
       final EngineLineMetrics line = lines[i];
       final List<RangeBox> boxes = line.boxes;
       final StringBuffer buffer = StringBuffer();
 
       int j = 0;
       while (j < boxes.length) {
         final RangeBox box = boxes[j++];
-        if (box is SpanBox && box.span == span) {
-          buffer.write(box.toText());
-          continue;
-        }
-
-        if (buffer.isNotEmpty) {
-          element.appendText(buffer.toString());
-          buffer.clear();
-        }
 
         if (box is SpanBox) {
-          span = box.span;
-          element = html.document.createElement('span') as html.HtmlElement;
+          lastSpanElement = html.document.createElement('span') as html.HtmlElement;
```

Comment:
Why not `html.SpanElement`?

Reply:
No reason. Is one different from the other?

I'll keep it as `html.document.createElement('span')` for now so I can easily change it to `flt-span` next.